libopenage/gamestate/game_main.cpp

```diff
@@ -22,28 +22,32 @@ GameMain::GameMain(const Generator &generator)
 	// players
 	unsigned int i = 0;
 	for (auto &name : generator.player_names()) {
-		this->players.emplace_back(this->add_civ(i), i, name);
+		this->players.push_back(new Player(this->add_civ(i), i, name));
```

Comment:
- Player is created at the start and destroyed at the end of the game, no need to increase complexity with unique or shared_ptr.
- I think the vector is already using the heap

> no polymorphism involved here

Can you explain this?